mmengine/runner/deepspeed_runner.py

```diff
@@ -0,0 +1,771 @@
+                '"model_wrapper" yet.')
+        return model
+
+    def _init_model_weights(self) -> None:
+        """Initialize the model weights when it is not initialized."""
+        if not self._weights_initialized:
+            super()._init_model_weights()
+            self._weights_initialized = True
+
+    def check_ds_config(self, ds_config):
+        """Check DeepSpeed configuration to prevent duplicated settings."""
+
+        # batch size
+        if ds_config.keys() & {
+                'train_batch_size', 'train_micro_batch_size_per_gpu',
+                'gradient_accumulation_steps'
+        }:
+            self.logger.warning(
+                'Batch size in DeepSpeed configuration will be ignored')
+
+        # optimizer
+        if 'optimizer' in ds_config.keys():
+            self.logger.warning(
+                'Optimizer in DeepSpeed configuration will be ignored')
+
+        # LR scheduler
+        if 'scheduler' in ds_config.keys():
+            self.logger.warning(
+                'LR scheduler in DeepSpeed configuration will be ignored')
+
+    def inject_basemodel_methods(self):
+        """inject methods from ``BaseModel`` into ``DeepSpeedEngine`` to make
+        ``DeepSpeedEngine`` support the ``train_step`` method appropriately.
+
+        Without injecting, ``DeepSpeedOptimWrapper`` tries ``backward`` from
+        ``BaseModel``, which should be in ``DeepSpeedEngine``.
+        """
+
+        def _train_step(self, data: Union[dict, tuple, list],
+                        optim_wrapper) -> Dict[str, torch.Tensor]:
+            with optim_wrapper.optim_context(self):
+                data = self.data_preprocessor(data, True)
+                losses = self._run_forward(data, mode='loss')  # type: ignore
+            parsed_losses, log_vars = self.parse_losses(losses)  # type: ignore
+            optim_wrapper.update_params(parsed_losses)
+            return log_vars
+
+        self.model.train_step = types.MethodType(_train_step, self.model)
+
+    def resume(
+        self,
+        filename: str,
+        resume_optimizer: bool = True,
+        resume_param_scheduler: bool = True,
+        map_location: Union[str, Callable] = 'default',
+    ) -> None:
+        """Resume model from checkpoint.
+
+        Args:
+            filename (str): Accept local filepath, URL, ``torchvision://xxx``,
+                ``open-mmlab://xxx``.
+            resume_optimizer (bool): Whether to resume optimizer state.
+                Defaults to True.
+            resume_param_scheduler (bool): Whether to resume param scheduler
+                state. Defaults to True.
+            map_location (str or callable):A string or a callable function to
+                specifying how to remap storage locations.
+                Defaults to 'default'.
+        """
+
+        if self.model.zero_optimization_partition_weights():
+            device = get_device()
+            checkpoint = _load_checkpoint(filename, map_location=device)
```

Comment:
https://github.com/microsoft/DeepSpeed/blob/211055216792cbb52ab6d355f698c194f9c55efb/deepspeed/runtime/engine.py#L2541

This logic tries to follow the main loading logic of `DeepSpeedEngine`.